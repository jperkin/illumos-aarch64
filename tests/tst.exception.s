/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.2.3 Exception generation
 */
.macro exception op, immopt=0
	.if \immopt
	\op
	.else
	\op #0
	.endif
	\op #65535
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	exception svc
	exception hvc
	exception smc
	exception brk
	exception hlt
	exception dcps1 1
	exception dcps2 1
	exception dcps3 1
.size libdis_test, [.-libdis_test]
