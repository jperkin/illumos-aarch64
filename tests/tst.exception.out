    libdis_test:      01 00 00 d4  svc #0x0
    libdis_test+0x4:  e1 ff 1f d4  svc #0xffff
    libdis_test+0x8:  02 00 00 d4  hvc #0x0
    libdis_test+0xc:  e2 ff 1f d4  hvc #0xffff
    libdis_test+0x10: 03 00 00 d4  smc #0x0
    libdis_test+0x14: e3 ff 1f d4  smc #0xffff
    libdis_test+0x18: 00 00 20 d4  brk #0x0
    libdis_test+0x1c: e0 ff 3f d4  brk #0xffff
    libdis_test+0x20: 00 00 40 d4  hlt #0x0
    libdis_test+0x24: e0 ff 5f d4  hlt #0xffff
    libdis_test+0x28: 01 00 a0 d4  dcps1
    libdis_test+0x2c: e1 ff bf d4  dcps1 #0xffff
    libdis_test+0x30: 02 00 a0 d4  dcps2
    libdis_test+0x34: e2 ff bf d4  dcps2 #0xffff
    libdis_test+0x38: 03 00 a0 d4  dcps3
    libdis_test+0x3c: e3 ff bf d4  dcps3 #0xffff
